bin/update_changelog.sh

```diff
@@ -0,0 +1,104 @@
+#!/bin/bash
+
+# Script to create a new RC entry
+#
+# Actions:
+# 1. find latest published version
+# 2. find the TBD planned big version
+# 3. create new RC version entry
+# 4. add change log entries
+
+set -e # exit on errors
+
+# TODO: provide that as parameters?
+
+CHANGE_FILE='CHANGES.rst'
+RC_DATE=$(date +'%m-%d-%Y')
+WORKSPACE_DIR="${GITHUB_WORKSPACE:-.}/rc"
+TMP_CHANGE_LOG="${WORKSPACE_DIR}/rc-${RC_DATE}.txt"
+
+
+############
+# CLEANUPS #
+############
+
+rm -rf $TMP_CHANGE_LOG
+mkdir -p $WORKSPACE_DIR
+
+
+##################
+# INITIALIZATION #
+##################
+
+echo "Updating $CHANGE_FILE:"
+
+LAST_VERSION=$(grep -m1 -E ' \([0-9]+-[0-9]+-[0-9]+\)$' $CHANGE_FILE | awk '{ print $1 }')
+
+echo "Detected last release version: $LAST_VERSION"
+
+
+###################
+# VERSION BUMPING #
+###################
+
+echo "Bumping patch version..."
+MAJOR_COLUMN=1
+MINOR_COLUMN=2
+PATCH_COLUMN=3
+
+BUMPED_VERSION=$(echo $LAST_VERSION | awk -F. -v K=$PATCH_COLUMN '{$K+=1; print $0}' OFS='.')
```

Comment:
nit: I _love_ awk and use it extensively, but I'll admit it's not always obvious what it's doing. Adding a comment here explaining that `-F` is setting the field separator for the input, that the argument to `-v` is setting the variable `K` for use in the expression, and that `OFS` is setting the field separator for the output might be helpful for new contributors, so that we wind up with e.g. `1.2.3` converted to `1.2.4`. Another way to help w/clarity would be to use the long-form arguments, i.e. `--field-separator` instead of `-F` and `--assign` instead of `-v`. I think this is not POSIX compliant, but it's easy enough to say that people have to have GNU awk to run this script.

Reply:
For now, I've decided to just provide the documentation in the comments :) I think the full names would have actually been nice, but I was a bit worried not to struggle too much with possible compatibility issues as it's intended to run on the GitHub Actions 😅 Will definitely revisit this after the new release gets public ✌️